javalib/src/main/scala/java/time/LocalDate.scala

```diff
@@ -0,0 +1,432 @@
+package java.time
+
+import scala.scalajs.js
+
+import java.time.chrono.{IsoEra, Era, IsoChronology, ChronoLocalDate}
+import java.time.temporal._
+
+final class LocalDate private (year: Int, month: Int, dayOfMonth: Int)
+    extends ChronoLocalDate with Serializable {
+  import Preconditions.requireDateTime
+  import ChronoField._
+  import ChronoUnit._
+  import LocalDate._
+
+  requireDateTime(year >= minYear && year <= maxYear,
+      s"Invalid value for year: $year")
+  requireDateTime(month > 0 && month <= 12, s"Invalid value for month: $month")
+
+  private val _month = Month.of(month)
+  private val _isLeapYear = iso.isLeapYear(year)
+
+  requireDateTime(dayOfMonth > 0 && dayOfMonth <= _month.maxLength,
+      s"Invalid value for dayOfMonth: $dayOfMonth")
+  requireDateTime(_isLeapYear || dayOfMonth <= _month.minLength,
+      s"Invalid value for dayOfMonth: $dayOfMonth")
+
+  private val dayOfYear = _month.firstDayOfYear(_isLeapYear) + dayOfMonth - 1
+  private val epochDay = {
+    val year1 = year - 1970
+    val daysBeforeYear = daysBeforeYears(MathJDK8Bridge.floorMod(year1, 400))._2
+    val offset =
+      MathJDK8Bridge.floorDiv(year1, 400).toLong * daysInFourHundredYears
+    offset + daysBeforeYear + dayOfYear - 1
+  }
+  private val dayOfWeek = MathJDK8Bridge.floorMod(epochDay + 3, 7).toInt + 1
+
+  // Implemented by ChronoLocalDate
+  // def isSupported(field: TemporalField): Boolean
+  // def isSupported(unit: TemporalUnit): Boolean
+
+  // Implemented by TemporalAccessor
+  // def range(field: TemporalField): ValueRange
+  // def get(field: TemporalField): Int
+
+  def getLong(field: TemporalField): Long = field match {
+    case DAY_OF_WEEK => dayOfWeek
+
+    case ALIGNED_DAY_OF_WEEK_IN_MONTH => (dayOfMonth - 1) % 7 + 1
+
+    case ALIGNED_DAY_OF_WEEK_IN_YEAR => (getLong(DAY_OF_YEAR) - 1) % 7 + 1
+
+    case DAY_OF_MONTH => dayOfMonth
+
+    case DAY_OF_YEAR => dayOfYear
+
+    case EPOCH_DAY => epochDay
+
+    case ALIGNED_WEEK_OF_MONTH => (dayOfMonth - 1) / 7 + 1
+
+    case ALIGNED_WEEK_OF_YEAR => (dayOfYear - 1) / 7 + 1
+
+    case MONTH_OF_YEAR => month
+
+    case PROLEPTIC_MONTH => year.toLong * 12 + month - 1
+
+    case YEAR_OF_ERA => if (year > 0) year else -year + 1
```

Comment:
Are you telling this for readability concerns, or just for performance concerns? If the latter, it's not necessary: the optimizer can rewrite `-year + 1` into `1 - year` on its own ;)